app/src/main/kotlin/compiler/intermediate_form/ControlFlow.kt

```diff
@@ -12,9 +18,178 @@ object ControlFlow {
+
+                            last = mapLinkType(conditionEnd, LinkType.CONDITIONAL_TRUE)
+                            processBlock(statement.actionWhenTrue)
+                            val trueBranchEnd = last
+
+                            last = mapLinkType(conditionEnd, LinkType.CONDITIONAL_FALSE)
+                            if (statement.actionWhenFalse != null)
+                                processBlock(statement.actionWhenFalse)
+                            val falseBranchEnd = last
+
+                            last = trueBranchEnd + falseBranchEnd
+                        }
+
+                        is Statement.Loop -> {
+                            val conditionEntry = addExpression(statement.condition, null)!!
+                            val conditionEnd = last
+
+                            val outerBreaking = breaking
+                            val outerContinuing = continuing
+
+                            breaking = mutableListOf()
+                            continuing = mutableListOf()
+
+                            last = mapLinkType(conditionEnd, LinkType.CONDITIONAL_TRUE)
+                            processBlock(statement.action)
+                            val end = last
+
+                            for (node in end + continuing!!)
+                                link(node, conditionEntry)
+
+                            last = mapLinkType(conditionEnd, LinkType.CONDITIONAL_FALSE) + breaking!!
+
+                            breaking = outerBreaking
+                            continuing = outerContinuing
+                        }
+
+                        is Statement.LoopBreak -> {
+                            if (breaking != null)
+                                breaking!!.addAll(last)
+                            else
+                                diagnostics.report(ControlFlowDiagnostic.BreakOutsideOfLoop(statement))
+
+                            last = emptyList()
+                        }
+
+                        is Statement.LoopContinuation -> {
+                            if (continuing != null)
+                                continuing!!.addAll(last)
+                            else
+                                diagnostics.report(ControlFlowDiagnostic.ContinuationOutsideOfLoop(statement))
+
+                            last = emptyList()
+                        }
+
+                        is Statement.FunctionReturn -> {
+                            addExpression(statement.value, null)
+
+                            last = emptyList()
+                        }
+                    }
+                }
+            }
+
+            processBlock(function.body)
+
+            controlFlowGraphs[function] = ControlFlowGraph(
+                treeRoots,
+                entryTreeRoot,
+                unconditionalLinks,
+                conditionalTrueLinks,
+                conditionalFalseLinks
+            )
+        }
+
+        for (global in program.globals) {
```

Comment:
How about this cool one-liner: `program.globals.filterIsInstance<Program.Global.FunctionDefinition>().forEach { processFunction(it.function) }`

Reply:
Sure.